src/deadline/job_attachments/api/manifest.py

```diff
@@ -54,80 +53,68 @@ def _manifest_snapshot(
-                full_diff_path = f"{root}/{diff_item[1].path}"
-                changed_paths.append(full_diff_path)
-                logger.echo(f"Found difference at: {full_diff_path}, Status: {diff_item[0]}")
+
+        # Fast comparison using time stamps and sizes.
+        if not force_rehash:
+            changed_paths = _fast_file_list_to_manifest_diff(
+                root, current_files, source_manifest, logger
+            )
+        else:
+            # In "slow / thorough" mode, we check by hash, which is definitive.
+            output_manifest = _create_manifest_for_single_root(
+                files=current_files, root=root, logger=logger
+            )
+            if not output_manifest:
+                return None
+            differences: List[Tuple[FileStatus, BaseManifestPath]] = compare_manifest(
+                source_manifest, output_manifest
+            )
+            for diff_item in differences:
+                if diff_item[0] == FileStatus.MODIFIED or diff_item[0] == FileStatus.NEW:
+                    full_diff_path = f"{root}/{diff_item[1].path}"
+                    changed_paths.append(full_diff_path)
+                    logger.echo(f"Found difference at: {full_diff_path}, Status: {diff_item[0]}")
+
+        # If there were no files diffed, return None, there was nothing to snapshot.
+        if len(changed_paths) == 0:
+            return None
 
         # Since the files are already hashed, we can easily re-use has_attachments to remake a diff manifest.
-        diff_group = asset_manager.prepare_paths_for_upload(
-            input_paths=changed_paths, output_paths=[root], referenced_paths=[]
-        )
-        _, diff_manifests = _hash_attachments(
-            asset_manager=asset_manager,
-            asset_groups=diff_group.asset_groups,
-            total_input_files=diff_group.total_input_files,
-            total_input_bytes=diff_group.total_input_bytes,
-            print_function_callback=logger.echo,
-            hashing_progress_callback=hash_callback_manager.callback,
+        output_manifest = _create_manifest_for_single_root(
+            files=changed_paths, root=root, logger=logger
         )
-        output_manifest = diff_manifests[0].asset_manifest
+        if not output_manifest:
+            return None
 
     # Write created manifest into local file, at the specified location at destination
     if output_manifest is not None:
-
+        # Encode the root path as
+        root_hash: str = hash_data(root.encode("utf-8"), output_manifest.get_default_hash_alg())
         timestamp = datetime.datetime.now().strftime("%Y-%m-%dT%H-%M-%S")
         manifest_name = name if name else root.replace("/", "_")
         manifest_name = manifest_name[1:] if manifest_name[0] == "_" else manifest_name
-        manifest_name = f"{manifest_name}-{timestamp}.manifest"
+        manifest_name = f"{manifest_name}-{root_hash}-{timestamp}.manifest"
```

Comment:
This would help the attachment CLI with path mapping, thank you for adding the hash!

Reply:
:)  Yes invent and simplify.